internal/internal.go

```diff
@@ -190,12 +190,19 @@ func (p *envoyExtAuthzGrpcServer) listen() {
 }
 
 func (p *envoyExtAuthzGrpcServer) Check(ctx context.Context, req *ext_authz.CheckRequest) (resp *ext_authz.CheckResponse, err error) {
+	if ctx.Err() != nil {
+		err = errors.Wrap(ctx.Err(), "server overload error")
+		return nil, err
+	}
+
```

Comment:
I think we should move this down to L225 so that we see this error in the decision log (as is, it would not be logged.) Also, in terms of wording, I'm not sure what the final message looks like (since we're using `errors.Wrap` here) but I'd like it to indicate that the check request was timed out before query execution started.

Reply:
Updated. The final error message looks something like  -> `check request timed out before query execution: context deadline exceeded`.

Reply:
LGTM!